tests/integration/source_code/solacc.py

```diff
@@ -63,17 +75,34 @@ def lint_all():
             parseable += 1
         except Exception as e:  # pylint: disable=broad-except
             # here we're most likely catching astroid & sqlglot errors
-            logger.error(f"Error during parsing of {file}: {e}".replace("\n", " "), exc_info=e)
+            if file_to_lint:
+                logger.error(f"Error during parsing of {file}: {e}".replace("\n", " "), exc_info=e)
+            else:
+                logger.error(f"Error during parsing of {file}: {e}".replace("\n", " "))
+            if file_to_lint is None:
+                # populate solacc-unparsed.txt
+                with unparsed.open(mode="a") as f:
+                    f.write(file.relative_to(dist).as_posix())
+                    f.write("\n")
     parseable_pct = int(parseable / len(all_files) * 100)
-    logger.info(f"Parseable: {parseable_pct}% ({parseable}/{len(all_files)}), missing imports: {missing_imports}")
+    logger.info(
+        f"Skipped: {len(skipped or [])}, parseable: {parseable_pct}% ({parseable}/{len(all_files)}), missing imports: {missing_imports}"
+    )
     if parseable_pct < 100:
         sys.exit(1)
 
 
-if __name__ == "__main__":
+def main(args: list[str]):
     install_logger()
     logging.root.setLevel(logging.INFO)
-    logger.info("Cloning...")
-    clone_all()
+    file_to_lint = args[1] if len(args) > 1 else None
+    if not file_to_lint:
+        # don't clone if linting just one file, assumption is we're troubleshooting
+        logger.info("Cloning...")
+        clone_all()
     logger.info("Linting...")
```

Comment:
Would like to see information about what's going to be linted here: there might not be any other feedback about what's happening until it's finished.

Reply:
This is a tool for us, so not sure how useful such feedback would be, but feel free to enhance